app/src/main/java/com/android/sample/model/recipe/DefaultRecipeRepository.kt

```diff
@@ -156,4 +147,13 @@ class DefaultRecipeRepository(
   override fun deleteAllDownloads(onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
     localRepository.deleteAll(onSuccess, onFailure)
   }
+
+  override fun filterSearch(
```

Comment:
**Adding `filterSearch`:**  
   Cool addition of `filterSearch`! Throwing `UnsupportedOperationException` for now makes sense if it’s not yet implemented. Just remember to revisit this once the functionality is ready, or maybe add a TODO comment to ensure it’s tracked.

Reply:
Ok this was the source of the crash therefore it is implemented now! 😄